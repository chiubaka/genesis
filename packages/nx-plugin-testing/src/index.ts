export * from "./setup";
export * from "./testingWorkspace";
export * from "./verdaccio";
