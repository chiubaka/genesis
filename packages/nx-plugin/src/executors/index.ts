export * from "./nxPluginE2e";
