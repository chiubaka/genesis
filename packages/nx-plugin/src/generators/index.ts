export * from "./linting";
export * from "./preset";
