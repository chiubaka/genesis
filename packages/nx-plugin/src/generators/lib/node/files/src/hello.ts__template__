import crypto from "node:crypto";
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export const hello = "Hello, world!";

export function writeHelloWorld() {
  const randomSuffix = crypto.randomBytes(48).toString("base64url");
  const fileName = `hello-${randomSuffix}.txt`;

  const tmpDir = os.tmpdir();
  const filePath = path.join(tmpDir, fileName);

  // eslint-disable-next-line security/detect-non-literal-fs-filename
  fs.writeFileSync(filePath, hello);

  return filePath;
}

export class Greeter {
  public hello() {
    return hello;
  }
}
