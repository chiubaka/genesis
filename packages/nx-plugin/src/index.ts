export * from "./generators";
export * from "./utils";
