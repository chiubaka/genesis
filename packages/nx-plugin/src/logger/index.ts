import { Logger, LoggerOptions, LogLevel } from "./Logger";
import { WinstonLogger } from "./winston";

export const createLogger = (
  name: string,
  options: Partial<LoggerOptions> = {},
): Logger => {
  return new WinstonLogger({
    level: LogLevel.Info,
    ...options,
    output: {
      console: true,
      file: [
        {
          path: `${name}.log`,
        },
      ],
      ...options.output,
    },
  });
};

export const generatorLogger = createLogger("generator");
