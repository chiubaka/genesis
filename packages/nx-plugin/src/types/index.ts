export * from "./codecov";
export * from "./dockerCompose";
export * from "./npm";
export * from "./nx";
export * from "./verdaccio";
export * from "./yarn";
