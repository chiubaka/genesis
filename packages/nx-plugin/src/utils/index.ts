export * from "./addDependenciesToPackageJson";
export * from "./exec";
export * from "./getProjectDir";
export * from "./git";
export * from "./github";
export * from "./Project";
export * from "./projectNameCasings";
export * from "./tasks";
