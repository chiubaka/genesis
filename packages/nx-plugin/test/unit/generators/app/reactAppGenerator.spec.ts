import { createTreeWithEmptyWorkspace } from "@chiubaka/nx-plugin-testing";
import { Tree } from "@nrwl/devkit";

import { Project, reactAppGenerator } from "../../../../src";
import {
  fileMatchesSnapshot,
  reactE2eProjectTestCases,
  reactProjectTestCases,
} from "../../../cases";

describe("reactAppGenerator", () => {
  let tree: Tree;
  let project: Project;
  let e2eProject: Project;

  const getProject = () => {
    return project;
  };

  const getE2eProject = () => {
    return e2eProject;
  };

  beforeAll(() => {
    tree = createTreeWithEmptyWorkspace();
    project = new Project(tree, "react-app", "application");
    e2eProject = new Project(tree, "react-app-e2e", "e2e");
  });

  reactProjectTestCases(getProject);

  beforeAll(async () => {
    await reactAppGenerator(tree, {
      name: "react-app",
    });
  });

  describe("workspace", () => {
    describe("storybook", () => {
      fileMatchesSnapshot(".storybook/main.ts", getProject);
    });
  });

  describe("sample code", () => {
    it("generates an App directory", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.srcPath("App"))).toBe(true);
    });

    it("generates an assets directory", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.srcPath("assets"))).toBe(true);
    });

    it("generates an environments directory", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.srcPath("environments"))).toBe(true);
    });

    it("generates a favicon", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.srcPath("favicon.ico"))).toBe(true);
    });

    fileMatchesSnapshot("index.html", getProject, (project: Project) =>
      project.srcPath("index.html"),
    );

    fileMatchesSnapshot("main.tsx", getProject, (project: Project) =>
      project.srcPath("main.tsx"),
    );

    fileMatchesSnapshot("polyfills.ts", getProject, (project: Project) =>
      project.srcPath("polyfills.ts"),
    );

    fileMatchesSnapshot("styles.scss", getProject, (project: Project) =>
      project.srcPath("styles.scss"),
    );

    describe("generates a sample component", () => {
      fileMatchesSnapshot("App/index.ts", getProject, (project: Project) =>
        project.srcPath("App/index.ts"),
      );

      fileMatchesSnapshot("App/App.tsx", getProject, (project: Project) =>
        project.srcPath("App/App.tsx"),
      );

      fileMatchesSnapshot(
        "App/App.module.scss",
        getProject,
        (project: Project) => project.srcPath("App/App.module.scss"),
      );

      fileMatchesSnapshot(
        "App/App.stories.tsx",
        getProject,
        (project: Project) => project.srcPath("App/App.stories.tsx"),
      );

      fileMatchesSnapshot("App/NxWelcome.tsx", getProject, (project: Project) =>
        project.srcPath("App/NxWelcome.tsx"),
      );

      fileMatchesSnapshot(
        "App/NxWelcome.stories.tsx",
        getProject,
        (project: Project) => project.srcPath("App/NxWelcome.stories.tsx"),
      );
    });

    describe("generates a sample unit test", () => {
      fileMatchesSnapshot("unit/App.spec.tsx", getProject, (project: Project) =>
        project.testPath("unit/App.spec.tsx"),
      );
    });

    describe("environments", () => {
      fileMatchesSnapshot(
        "environments/environment.ts",
        getProject,
        (project: Project) => project.srcPath("environments/environment.ts"),
      );

      fileMatchesSnapshot(
        "environments/environment.prod.ts",
        getProject,
        (project: Project) =>
          project.srcPath("environments/environment.prod.ts"),
      );
    });
  });

  describe("E2E project", () => {
    reactE2eProjectTestCases(getE2eProject);

    describe("sample tests", () => {
      fileMatchesSnapshot("e2e/App.cy.ts", getE2eProject, (project: Project) =>
        project.srcPath("e2e/App.cy.ts"),
      );

      fileMatchesSnapshot(
        "e2e/App/App.cy.ts",
        getE2eProject,
        (project: Project) => project.srcPath("e2e/App/App.cy.ts"),
      );

      fileMatchesSnapshot(
        "e2e/NxWelcome/NxWelcome.cy.ts",
        getE2eProject,
        (project: Project) => project.srcPath("e2e/NxWelcome/NxWelcome.cy.ts"),
      );
    });
  });
});
