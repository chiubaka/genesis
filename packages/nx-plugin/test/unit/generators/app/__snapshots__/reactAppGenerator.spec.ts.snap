// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactAppGenerator .babelrc matches snapshot 1`] = `
Object {
  "plugins": Array [],
  "presets": Array [
    Array [
      "@nrwl/react/babel",
      Object {
        "runtime": "automatic",
      },
    ],
  ],
}
`;

exports[`reactAppGenerator .browserslistrc matches snapshot 1`] = `
"# This file is used by:
# 1. autoprefixer to adjust CSS to support the below specified browsers
# 2. babel preset-env to adjust included polyfills
#
# For additional information regarding the format and rule options, please see:
# https://github.com/browserslist/browserslist#queries
#
# If you need to support different browsers in production, you may tweak the list below.

last 1 Chrome version
last 1 Firefox version
last 2 Edge major versions
last 2 Safari major version
last 2 iOS major versions
Firefox ESR
not IE 9-11 # For IE 9-11 support, remove 'not'."
`;

exports[`reactAppGenerator .eslintrc.json matches snapshot 1`] = `
Object {
  "extends": Array [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": Array [
    "node_modules",
  ],
}
`;

exports[`reactAppGenerator E2E project .eslintrc.json matches snapshot 1`] = `
Object {
  "extends": Array [
    "plugin:cypress/recommended",
    "../../.eslintrc.json",
  ],
  "ignorePatterns": Array [
    "node_modules",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
        "*.tsx",
        "*.js",
        "*.jsx",
      ],
      "rules": Object {},
    },
    Object {
      "files": Array [
        "src/plugins/index.js",
      ],
      "rules": Object {
        "@typescript-eslint/no-var-requires": "off",
        "no-undef": "off",
      },
    },
  ],
}
`;

exports[`reactAppGenerator E2E project cypress.config.ts matches snapshot 1`] = `
"import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: nxE2EPreset(__dirname)
});
"
`;

exports[`reactAppGenerator E2E project project.json matches snapshot 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "implicitDependencies": Array [
    "react-app",
  ],
  "projectType": "application",
  "sourceRoot": "e2e/react-app-e2e/src",
  "tags": Array [],
  "targets": Object {
    "e2e": Object {
      "configurations": Object {
        "production": Object {
          "devServerTarget": "react-app:serve:production",
        },
      },
      "executor": "@nrwl/cypress:cypress",
      "options": Object {
        "cypressConfig": "e2e/react-app-e2e/cypress.config.ts",
        "devServerTarget": "react-app:serve:development",
        "testingType": "e2e",
      },
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "e2e/react-app-e2e/**/*.{js,ts}",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
  },
}
`;

exports[`reactAppGenerator E2E project tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "cypress",
      "node",
    ],
  },
  "extends": "../../tsconfig.base.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
    "cypress.config.ts",
  ],
}
`;

exports[`reactAppGenerator jest.config.ts matches snapshot 1`] = `
"export default {
  displayName: \\"react-app\\",
  preset: \\"../../jest.preset.js\\",
  testEnvironment: \\"jsdom\\",
  transform: {
    "^(?!.*\\.(js|jsx|ts|tsx|css|json)$)": "@nrwl/react/plugins/jest",
    "^.+\\.[tj]sx?$": "babel-jest",
  },
  moduleFileExtensions: [\\"ts\\", \\"tsx\\", \\"js\\", \\"jsx\\"],
  coverageDirectory: \\"../../reports/coverage/packages/react-app\\",
  reporters: [
    \\"default\\",
    [
      \\"jest-junit\\",
      {
        addFileAttribute: \\"true\\",
        classNameTemplate: \\"{suitename}\\",
        outputDirectory: \\"reports/junit\\",
        outputName: \\"react-app.xml\\",
      },
    ],
  ],
};
"
`;

exports[`reactAppGenerator project.json matches snapshot 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "application",
  "sourceRoot": "packages/react-app/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "configurations": Object {
        "development": Object {
          "extractLicenses": false,
          "optimization": false,
          "sourceMap": true,
          "vendorChunk": true,
        },
        "production": Object {
          "extractLicenses": true,
          "fileReplacements": Array [
            Object {
              "replace": "packages/react-app/src/environments/environment.ts",
              "with": "packages/react-app/src/environments/environment.prod.ts",
            },
          ],
          "namedChunks": false,
          "optimization": true,
          "outputHashing": "all",
          "sourceMap": false,
          "vendorChunk": false,
        },
      },
      "defaultConfiguration": "production",
      "executor": "@nrwl/web:webpack",
      "options": Object {
        "assets": Array [
          "packages/react-app/src/favicon.ico",
          "packages/react-app/src/assets",
        ],
        "baseHref": "/",
        "compiler": "babel",
        "index": "packages/react-app/src/index.html",
        "main": "packages/react-app/src/main.tsx",
        "outputPath": "dist/packages/react-app",
        "polyfills": "packages/react-app/src/polyfills.ts",
        "scripts": Array [],
        "styles": Array [
          "packages/react-app/src/styles.scss",
        ],
        "tsConfig": "packages/react-app/tsconfig.app.json",
        "webpackConfig": "@nrwl/react/plugins/webpack",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "build-storybook": Object {
      "configurations": Object {
        "ci": Object {
          "quiet": true,
        },
      },
      "executor": "@nrwl/storybook:build",
      "options": Object {
        "config": Object {
          "configFolder": "packages/react-app/.storybook",
        },
        "outputPath": "dist/storybook/react-app",
        "uiFramework": "@storybook/react",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "packages/react-app",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
    "serve": Object {
      "configurations": Object {
        "development": Object {
          "buildTarget": "react-app:build:development",
        },
        "production": Object {
          "buildTarget": "react-app:build:production",
          "hmr": false,
        },
      },
      "defaultConfiguration": "development",
      "executor": "@nrwl/web:dev-server",
      "options": Object {
        "buildTarget": "react-app:build",
        "hmr": true,
      },
    },
    "storybook": Object {
      "configurations": Object {
        "ci": Object {
          "quiet": true,
        },
      },
      "executor": "@nrwl/storybook:storybook",
      "options": Object {
        "config": Object {
          "configFolder": "packages/react-app/.storybook",
        },
        "port": 4400,
        "uiFramework": "@storybook/react",
      },
    },
    "test": Object {
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "packages/react-app/jest.config.ts",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "reports/coverage/packages/react-app",
        "reports/junit/react-app.xml",
      ],
    },
  },
}
`;

exports[`reactAppGenerator storybook .storybook/main.ts matches snapshot 1`] = `
"import { rootMain } from '../../../.storybook/main';
import type { StorybookConfig, Options } from '@storybook/core-common';


const config: StorybookConfig = {
  ...rootMain,
  
  core: { ...rootMain.core, builder: 'webpack5' },
  

  stories: [
    ...rootMain.stories,
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)'
  ],
  addons: [...(rootMain.addons || []) , '@nrwl/react/plugins/storybook' 
    
    
  ],
  webpackFinal: async (config, { configType }: Options) => {
    // apply any global webpack configs that might have been specified in .storybook/main.ts
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, { configType } as Options);
    }

    // add your own webpack tweaks if needed

    return config;
  },
};

module.exports = config;
"
`;

exports[`reactAppGenerator storybook .storybook/preview.ts matches snapshot 1`] = `""`;

exports[`reactAppGenerator storybook .storybook/tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "emitDecoratorMetadata": true,
    "outDir": "",
  },
  "exclude": Array [
    "../**/*.spec.ts",
    "../**/*.spec.js",
    "../**/*.spec.tsx",
    "../**/*.spec.jsx",
  ],
  "extends": "../tsconfig.json",
  "files": Array [
    "../../../node_modules/@nrwl/react/typings/styled-jsx.d.ts",
    "../../../node_modules/@nrwl/react/typings/cssmodule.d.ts",
    "../../../node_modules/@nrwl/react/typings/image.d.ts",
  ],
  "include": Array [
    "../src/**/*.stories.ts",
    "../src/**/*.stories.js",
    "../src/**/*.stories.jsx",
    "../src/**/*.stories.tsx",
    "../src/**/*.stories.mdx",
    "*.ts",
    "*.js",
  ],
}
`;

exports[`reactAppGenerator tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "declaration": true,
  },
  "exclude": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.spec.tsx",
    "**/*.test.tsx",
    "test",
    "**/*.stories.ts",
    "**/*.stories.js",
    "**/*.stories.jsx",
    "**/*.stories.tsx",
  ],
  "extends": "./tsconfig.json",
  "include": Array [
    "**/*.ts",
    "**/*.tsx",
  ],
}
`;

exports[`reactAppGenerator tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "jsx": "react-jsx",
    "lib": Array [
      "dom",
      "dom.iterable",
      "esnext",
    ],
    "module": "esnext",
    "outDir": "../../dist/out-tsc",
    "target": "es5",
  },
  "exclude": Array [
    "**/*.stories.ts",
    "**/*.stories.js",
    "**/*.stories.jsx",
    "**/*.stories.tsx",
  ],
  "extends": "../../tsconfig.base.json",
  "files": Array [
    "../../node_modules/@nrwl/react/typings/cssmodule.d.ts",
    "../../node_modules/@nrwl/react/typings/image.d.ts",
  ],
  "include": Array [],
  "references": Array [
    Object {
      "path": "./tsconfig.app.json",
    },
    Object {
      "path": "./tsconfig.spec.json",
    },
    Object {
      "path": "./.storybook/tsconfig.json",
    },
  ],
}
`;

exports[`reactAppGenerator tsconfig.spec.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "module": "commonjs",
    "types": Array [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.spec.tsx",
    "**/*.test.tsx",
    "**/*.d.ts",
    "test",
  ],
}
`;
